test/test_plugin.sh.in

```diff
@@ -0,0 +1,152 @@
+        PLUGINS_FOR_DIR2="$FROM_DIR/cygfilter_plugin2* $FROM_DIR/cygfilter_plugin4*"
+        ;;
+    *)
+        NULL_VFD_PLUGIN="$FROM_DIR/libnull_vfd_plugin*"
+        NULL_VOL_PLUGIN="$FROM_DIR/libnull_vol_connector*"
+        PLUGINS_FOR_DIR1="$FROM_DIR/libfilter_plugin1* $FROM_DIR/libfilter_plugin3*"
+        PLUGINS_FOR_DIR2="$FROM_DIR/libfilter_plugin2* $FROM_DIR/libfilter_plugin4*"
+        ;;
+esac
+
+# Directories where we'll copy plugins
+TEMP_PLUGIN_DIR=temp_plugins
+TEMP_FILTER_DIR1=temp_filter_plugin_dir1
+TEMP_FILTER_DIR2=temp_filter_plugin_dir2
+
+# Function to print a line-line message left justified in a field of
+# 70 characters beginning with the word "Testing".
+#
+TESTING() {
+    SPACES="                                                               "
+    echo "Testing $* $SPACES" | cut -c1-70 | tr -d '\012'
+}
+
+#############
+# Main Body #
+#############
+
+# Create plugin directories
+test -d $TEMP_PLUGIN_DIR || mkdir -p $TEMP_PLUGIN_DIR
+if [ $? != 0 ]; then
+    echo "Failed to create plugin test directory ($TEMP_PLUGIN_DIR)"
+    exit $EXIT_FAILURE
+fi
+test -d $TEMP_FILTER_DIR1 || mkdir -p $TEMP_FILTER_DIR1
+if [ $? != 0 ]; then
+    echo "Failed to create filter plugin test directory ($TEMP_FILTER_DIR1)"
+    exit $EXIT_FAILURE
+fi
+test -d $TEMP_FILTER_DIR2 || mkdir -p $TEMP_FILTER_DIR2
+if [ $? != 0 ]; then
+    echo "Failed to create filter plugin test directory ($TEMP_FILTER_DIR2)"
+    exit $EXIT_FAILURE
+fi
+
+# Copy plugins for the tests
+$CP $NULL_VFD_PLUGIN $TEMP_PLUGIN_DIR
+if [ $? != 0 ]; then
+    echo "Failed to copy NULL VFD plugin ($NULL_VFD_PLUGIN) to test directory."
+    exit $EXIT_FAILURE
+fi
+$CP $NULL_VOL_PLUGIN $TEMP_PLUGIN_DIR
+if [ $? != 0 ]; then
+    echo "Failed to copy NULL VOL plugin ($NULL_VOL_PLUGIN) to test directory."
+    exit $EXIT_FAILURE
+fi
+$CP $PLUGINS_FOR_DIR1 $TEMP_FILTER_DIR1
+if [ $? != 0 ]; then
+    echo "Failed to copy filter plugins ($PLUGINS_FOR_DIR1) to test directory."
+    exit $EXIT_FAILURE
+fi
+$CP $PLUGINS_FOR_DIR2 $TEMP_FILTER_DIR2
+if [ $? != 0 ]; then
+    echo "Failed to copy filter plugins ($PLUGINS_FOR_DIR2) to test directory."
+    exit $EXIT_FAILURE
+fi
+
+# Set plugin path
+ENVCMD="env HDF5_PLUGIN_PATH=${TEMP_PLUGIN_DIR}:${TEMP_FILTER_DIR1}:${TEMP_FILTER_DIR2}"
+
+# Run the tests
+$ENVCMD $FILTER_TEST_BIN
+if [ $? != 0 ]; then
+    nerrors=`expr $nerrors + 1`
+fi
+$ENVCMD $VFD_TEST_BIN
+if [ $? != 0 ]; then
+    nerrors=`expr $nerrors + 1`
+fi
+$ENVCMD $VOL_TEST_BIN
+if [ $? != 0 ]; then
+    nerrors=`expr $nerrors + 1`
+fi
+############################################
+# HDFFV-9655 test for relative path disabled
+# setup filter plugin path relative to test
+# actual executable is in the .libs folder
+#ENVCMD="env HDF5_PLUGIN_PATH=@/../${TEMP_FILTER_DIR1}:@/../${TEMP_FILTER_DIR2}"
+#
+# Run the test
+#$ENVCMD $FILTER_TEST_BIN
+#if [ $? != 0 ]; then
+#    nerrors=`expr $nerrors + 1`
+#fi
+#############################################
```

Comment:
I don't know what the deal is with this test. Should we fix this?

Reply:
That test isn't even in CMake.